### PR TITLE
Split nightly workflow by backend (IREE / SHARK)

### PR DESCRIPTION
**Pull request by monorimet**
Nightly Release assets are only published from the SHARK job.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -62,6 +63,7 @@ jobs:
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --exclude shark.venv,lit.cfg.py 
     - name: Build and validate the IREE package
+      if: matrix.backend == IREE
```
Comment (powderluv) [inline .github/workflows/nightly.yml:66]: Do you need double curly braces ?

Comment (monorimet) [reply .github/workflows/nightly.yml]: Quotes were missing. YML format is funky!

Comment (powderluv) [reply .github/workflows/nightly.yml]: I thought we needed this kind: https://github.com/llvm/torch-mlir/blob/068eec41cb959278d269cd1d307cd41c9a4b4c0c/.github/workflows/buildAndTest.yml#L94

**Review by powderluv — APPROVED** [review]
minor nit on using:
`if: {{ matrix.backend == 'SHARK' }}`

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (monorimet) [reply .github/workflows/nightly.yml]: Oh OK. I'll add that in while I fix the SHARK build and validate job

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -95,25 +92,20 @@ jobs:
...
-          tee -a pytest_results.txt
-        pytest -k 'vulkan' --ignore=benchmarks/tests/test_hf_benchmark.py --ignore=benchmarks/tests/test_benchmark.py --ignore=shark/tests/test_shark_importer.py --ignore=tank/tf/ |
+        pytest tank |
           tail -n 1 |
           tee -a pytest_results.txt
         if !(grep -Fxq " failed" pytest_results.txt) 
           then 
             export SHA=$(git log -1 --format='%h')
-            gsutil -m cp -r $GITHUB_WORKSPACE/gen_shark_tank/* gs://shark_tank/$SHA
+            gsutil -m cp -r ~/.local/shark_tank/* gs://shark_tank/$SHA
```
Comment (powderluv) [inline .github/workflows/nightly.yml:101]: Do we delete this directory at the beginning of the run?

Comment (powderluv) [reply .github/workflows/nightly.yml:101]: Also if I login and run something it will pollute the local tank during upload

**Review by powderluv — DISMISSED** [review]
Please verify it works after the merge

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (monorimet) [reply .github/workflows/nightly.yml]: Made a workaround so our CI shark_tank generates into the github workspace.

**Review by pashu123 — APPROVED** [review]
LGTM!

**Pull request merged**